### PR TITLE
Allow operator changes on category fields

### PR DESCRIPTION
**Pull request by iethree**
## Problem

On bulk filter select inputs, we had the strange situation where you could _see_ the operator next to the field name, but had to _edit_ it in the modal.

Screen Shot 2022-07-15 at 2 25 11 PM

These inputs appear in two cases:
1. Where a category input has more than 15 options, or
2. As a fallback for any field type other than dates, numbers, categories, or text

This exposes a bit of a larger issue: we handle category pickers in two different ways based on how many options they had

case | image
--- | ---
Small numbers of options got all their options inlined | Screen Shot 2022-07-15 at 2 27 27 PM
large numbers of options got the popover picker | Screen Shot 2022-07-15 at 2 27 16 PM

1. It's a little strange to get more operator options (like starts with/ends with) with the same type of field just when there's more options
3. It's a little strange to get inputs in a popover that we inline elsewhere (like starts with / ends with)

## Solution

1. Allow users to change operators on all category inputs
2. Show text input fields inline even for large sets of input values, so now the only case where a user gets a popover is for a large list of category values

categoryPickerDemo

### CODE REVIEW

File: frontend/test/__support__/e2e/helpers/e2e-bi-basics-helpers.js [hunk 1]
```diff
@@ -31,8 +37,8 @@ export function filterFieldPopover(fieldName, { value, placeholder } = {}) {
   return popover();
 }
 
-function getFilterField(fieldName) {
-  return cy.findByTestId(`filter-field-${fieldName}`);
+function getFilterField(fieldName, order = 0) {
+  return cy.findAllByTestId(`filter-field-${fieldName}`).eq(order);
 }
```
Comment (iethree) [inline frontend/test/__support__/e2e/helpers/e2e-bi-basics-helpers.js:42]: this adds the ability to easily test the Nth filter on the same field

File: frontend/src/metabase/query_builder/components/filters/modals/InlineCategoryPicker/InlineCategoryPicker.unit.spec.tsx [hunk 2]
```diff
@@ -354,14 +359,49 @@ describe("InlineCategoryPicker", () => {
...
+      const testFilter = new Filter(
+        ["contains", ["field", field.id, null], undefined],
+        null,
+        query,
+      );
+      const changeSpy = jest.fn();
+      const fetchSpy = jest.fn();
+
+      render(
+        <Provider store={store}>
```
Comment (WiNloSt) [inline frontend/src/metabase/query_builder/components/filters/modals/InlineCategoryPicker/InlineCategoryPicker.unit.spec.tsx:389]: What do you think about using `renderWithProviders`, which seems more established instead? https://github.com/metabase/metabase/blob/43f084bf3cbca42abdd78ed9c78c58fcda380dec/frontend/test/__support__/ui.js#L42

**Review by WiNloSt — APPROVED** [review]
This PR looks so clean, good job 👍

File: frontend/src/metabase/query_builder/components/filters/modals/BulkFilterList/BulkFilterList.tsx [hunk 3]
```diff
@@ -17,12 +17,7 @@ import Filter from "metabase-lib/lib/queries/structured/Filter";
 import { BulkFilterItem } from "../BulkFilterItem";
 import { SegmentFilterSelect } from "../BulkFilterSelect";
 import { InlineOperatorSelector } from "../InlineOperatorSelector";
-import {
-  ListRoot,
-  ListRow,
-  ListRowLabel,
```
[the remark this answers is not in the document]
Comment (iethree) [reply frontend/src/metabase/query_builder/components/filters/modals/BulkFilterList/BulkFilterList.tsx:23]: thanks. shouldn't the linter catch this?

File: frontend/src/metabase/query_builder/components/filters/modals/InlineCategoryPicker/InlineCategoryPicker.unit.spec.tsx (hunk 2 above)
Comment (iethree) [reply frontend/src/metabase/query_builder/components/filters/modals/InlineCategoryPicker/InlineCategoryPicker.unit.spec.tsx:389]: this is lovely! I didn't know about it!

**Pull request merged**